src/internal.c

```diff
@@ -222,6 +222,191 @@ static int SSL_hmac(WOLFSSL* ssl, byte* digest, const byte* in, word32 sz,
     int tsip_generatePremasterSecret();
     int tsip_generateEncryptPreMasterSecret(WOLFSSL *ssl, byte *out, word32 *outSz);
 #endif
+#if defined(OPENSSL_EXTRA) && defined(HAVE_SECRET_CALLBACK)
+
+    static void ToHexString(const byte* bin, byte* hex, int BinSz);
+    static int  SessionSecret_callback(WOLFSSL* ssl, void* secret,
+                                                  int* secretSz, void* ctx);
+    static int  SessionSecret_callback_Tls13(WOLFSSL* ssl, int id,
+                       const unsigned char* secret, int secretSz, void* ctx);
+
+    /* Convert input byte array to Hex string then add a null-terminator
+     *
+     * The buffer for hex must be equal or greater than (bin size * 2 + 1).
+     *
+     */
+    static void ToHexString(const byte* bin, byte* hex, int BinSz)
```

Comment:
As @JacobBarthelmeh pointed out in PR #3794, can we use Base16_Encode() here in place of ToHexString()?

Reply:
Replaced ToHexString with Base16_Encode.